ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -312,6 +333,50 @@ mkPeers h as =
 pointSchedule :: [Tick] -> NonEmpty PeerId -> Maybe PointSchedule
 pointSchedule ticks nePeerIds = (`PointSchedule` nePeerIds) <$> nonEmpty ticks
 
+-- | Convert a @SinglePeer@ schedule to a 'NodeState' schedule.
+--
+-- Accumulates the new points in each tick into the previous state, starting with a set of all
+-- 'Origin' points.
+--
+-- Also shifts all tick start times so that the first tip point is announced at the very beginning
+-- of the test, keeping the relative delays of the schedule intact.
+-- This is a preliminary measure to make the long range attack test work, since that relies on the
+-- honest node sending headers later than the adversary, which is not possible if the adversary's
+-- first tip point is delayed by 20 or more seconds due to being in a later slot.
+peerStates :: PeerId -> [(DiffTime, SchedulePoint)] -> [(DiffTime, Peer NodeState)]
```

Comment:
I feel this doesn't really belong in `Run` but rather in `PointSchedule`. To me it looks like this is just a conversion function between two different implementation of equivalent point schedules (the “single-peer” or “point-wise” ones or whatnot vs. the “table” ones). But we could also go for the policy that “point-wise” point schedules are _the_ point schedules and the table ones are just an intermediary representation used by the peer simulator?

Reply:
this is in `PointSchedule`, did you mean something else?